src/storage/irc_schema_set.cpp

```diff
@@ -11,12 +11,26 @@ namespace duckdb {
 IRCSchemaSet::IRCSchemaSet(Catalog &catalog) : catalog(catalog) {
 }
 
+// we are sure the schema does not exist, we performed the head request and got a 404 response.
+static bool SchemaDoesNotExist(case_insensitive_set_t &known_non_existent_schemas, const string &name) {
+	return known_non_existent_schemas.find(name) != known_non_existent_schemas.end();
+}
+
 optional_ptr<CatalogEntry> IRCSchemaSet::GetEntry(ClientContext &context, const string &name,
                                                   OnEntryNotFound if_not_found) {
 	lock_guard<mutex> l(entry_lock);
 	auto &ic_catalog = catalog.Cast<IRCatalog>();
 
+	auto &irc_transaction = IRCTransaction::Get(context, catalog);
+
+	auto verify_existence = irc_transaction.looked_up_entries.insert(name).second;
 	auto entry = entries.find(name);
+	if (entry != entries.end()) {
+		return entry->second.get();
+	}
+	if (!verify_existence) {
+		return nullptr;
```

Comment:
One thing wrong with my suggestion: this needs to take `if_not_found` into account, if `if_not_found != OnEntryNotFound::RETURN_NULL` this should probably throw

Reply:
Wouldn't the first request of the transaction already throw? Ah, maybe IF_NOT_FOUND has different values in the transaction.